src/storage/lsm.rs

```diff
@@ -210,20 +217,74 @@ impl LSM {
     }
 
     fn compact_sstables(&mut self) -> Result<()> {
-        let mut dense_idx = Memtable::default();
-        for old_sst in self.sstables.iter() {
-            dense_idx.update_from_commit_log(&old_sst.path)?;
+
+        let path = new_path(SSTABLES_DIR_PATH);
+        let mut file = OpenOptions::new()
+            .create(true)
+            .write(true)
+            .truncate(true)
+            .open(&path)?;
+
+        // TODO(btc): for each old SSTable file, create iterator
+        // TODO(btc): while there exists an iterator with values, loop over them selecting the last instance of the smallest key.
```

Comment:
This works, but would it be faster if we use a priority queue to pick the iterator?

Option 1: The priority queue is sorted by the tuple `(key asc, file timestamp asc, iter)`, and if the front of queue has a key that we've seen before, then we override the value in the merged vector.

Option 2: The priority queue is sorted by the tuple `(key asc, file time desc, iter)`, and if the front of the queue has a key that we've seen before, then we ignore this front item.

Not sure how hard this would be in rust.

Reply:
> This works, but would it be faster if we use a priority queue to pick the iterator?

maybe!
addressed in #18